src/components/CheckIn/CheckInDetails.js

```diff
@@ -0,0 +1,238 @@
+import React, { Component } from 'react';
+import { FormattedMessage } from 'react-intl';
+import PropTypes from 'prop-types';
+
+import { get } from 'lodash';
+
+import {
+  Callout,
+  Checkbox,
+  KeyValue,
+  Modal,
+  MultiColumnList,
+  Select,
+  TextField,
+} from '@folio/stripes/components';
+
+import getLocationsForSelect from '../Utils/getLocationsForSelect';
+import { fetchItems } from '../Receiving/util';
+import {
+  CHECKIN,
+  ITEMS,
+  LOCATIONS,
+} from '../Utils/resources';
+import { ITEM_STATUS } from '../Receiving/const';
+import CHECKIN_URLS from './const';
+import CheckInDetailsFooter from './CheckInDetailsFooter';
+import checkInItems from './util';
+
+import css from './CheckInDetails.css';
+
+class CheckInDetails extends Component {
+  static manifest = Object.freeze({
+    checkIn: CHECKIN,
+    items: ITEMS,
+    locations: LOCATIONS,
+    query: {},
+  })
+
+  static propTypes = {
+    close: PropTypes.func.isRequired,
+    location: PropTypes.object,
+    mutator: PropTypes.object,
+    items: PropTypes.arrayOf(PropTypes.object).isRequired,
+    resources: PropTypes.object,
+  }
+
+  constructor(props) {
+    super(props);
+    this.callout = React.createRef();
+
+    this.state = {
+      isAllChecked: true,
+      isLoading: true,
+      items: this.props.items,
+    };
+  }
+
+  componentDidMount() {
+    const { mutator, items } = this.props;
+
+    fetchItems(mutator, items).then(itemsMap => this.setState((state) => {
```

Comment:
I don't think you're required to use `this.setState((state) => {` here. And maybe we shouldn't mutate `pieces`, just save `itemsMap` in state and work with that. @NikitaSedyx what do you think?

Reply:
OK, let's leave it as is: state should contain `items`, which are the UI representation of mixin `piece` and `inventory item`. Props should contain `pieces` to make sense of naming.